core/Http/ControllerResolver.php

```diff
@@ -40,30 +40,18 @@ public function __construct(FactoryInterface $abstractFactory)
      */
     public function getController($module, $action, array &$parameters)
     {
-        $controller = $this->createPluginController($module, $action);
-        if ($controller) {
-            return $controller;
-        }
-
-        $controller = $this->createWidgetController($module, $action, $parameters);
-        if ($controller) {
-            return $controller;
-        }
-
-        $controller = $this->createReportController($module, $action, $parameters);
-        if ($controller) {
-            return $controller;
-        }
-
-        $controller = $this->createReportMenuController($module, $action, $parameters);
-        if ($controller) {
-            return $controller;
+        foreach(['Plugin', 'Widget', 'Report', 'ReportMenu'] as $controler) {
```

Comment:
Short array notation is not compatible with PHP 5.3, but Piwik 2.x still is. We can do it that way for Piwik 3.x